File: RedSheet.DbEntities/Models/PowerTypeDetail.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("PowerTypeDetails", Schema = "dbo")]
	[Serializable]
	public partial class PowerTypeDetail  
	{
 
	[Required]
	public bool IsOurKnowledge { get; set; } 
 
	[Required]
	public bool IsTheirKnowledge { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int PowerTypeDetailId { get; set; } 
 
	[Required] 
	[MaxLength(500)]
	public string Rationale { get; set; } 
	///<summary>
    ///{{PowerProjection}}
    ///</summary> 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public PowerProjection ActualId { get; set; } 
 
	[RelationshipTableAttribue("PowerTypes","dbo")]
	//Relationships
	public int PowerTypeId { get; set; } 
 
	[ForeignKey("PowerTypeId")]
	public PowerType PowerType { get; set; } 
	///<summary>
    ///{{PowerProjection}}
    ///</summary> 
	[RelationshipTableAttribue("ApplicationObjects","dbo")]
	public PowerProjection ProjectedId { get; set; } 
 
	[RelationshipTableAttribue("ProjectPowers","dbo")]
	//Relationships
	public int ProjectPowerId { get; set; } 
 
	[ForeignKey("ProjectPowerId")]
	public ProjectPower ProjectPower { get; set; } 

	public PowerTypeDetail()
	{
    }

    }
}

File: redsheet-portal/src/app/database-models/power-type-detail.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { PowerType, ProjectPower,  vPowerTypeDetailRecord  } from './'
export class PowerTypeDetail {
    constructor(powerTypeDetail?: PowerTypeDetail  | vPowerTypeDetailRecord )  {
        let properties = [ "isOurKnowledge", "isTheirKnowledge", "powerTypeDetailId", "rationale", "actualId", "powerTypeId", "projectedId", "projectPowerId",];
        for (let property of properties)
            if (powerTypeDetail && powerTypeDetail[property])
                this[property] = powerTypeDetail[property];
    }
 
	isOurKnowledge : boolean = false ;
 
	isTheirKnowledge : boolean = false ;
 
	powerTypeDetailId : number =   0 ;
 
    @required()
    @maxLength(500)
	rationale : string =   undefined;
 
    @range(0,2147483647)
	actualId : number =   undefined;
 
    @range(0,2147483647)
	powerTypeId : number =   undefined;
	powerType : PowerType  ;
 
    @range(0,2147483647)
	projectedId : number =   undefined;
 
    @range(0,2147483647)
	projectPowerId : number =   undefined;
	projectPower : ProjectPower  ;


}
